invenio_circulation/transitions/base.py

```diff
@@ -131,6 +131,7 @@ def before(self, loan, **kwargs):
 
         kwargs.setdefault('transaction_date', datetime.now())
         kwargs['transaction_date'] = parse_date(kwargs['transaction_date'])
+        self.prev_loan = loan.copy()
```

Comment:
Here you should deep copy because dict.copy() does a shallow copy. So, e.g 
```
a = dict(a=dict(b,1))
b = a.copy()
b['a']['b']=2
Output: 
b -> {'a': {'b': 3}}
a-> {'a': {'b': 3}}
```